src/Veldrid.SPIRV/SpirvCompilation.cs

```diff
@@ -357,6 +400,38 @@ public static class SpirvCompilation
             }
         }
 
+        /// <summary>
+        /// Composes the given major, minor, and patch values into a value that can be used as a TargetVersion
+        /// </summary>
+        /// <param name="major">MSL major version</param>
+        /// <param name="minor">MSL minor version</param>
+        /// <param name="patch">MSL patch version</param>
+        /// <returns></returns>
+        public static uint MakeMSLVersion(uint major, uint minor = 0, uint patch = 0)
```

Comment:
I didn't look to closely at the details of the various version formats, but I'm not really content with having this dangling off the API like this.

Would you rather see something more strongly typed like `class TargetVersion(uint major, uint minor, uint patch)` exposed to callers in the public interface, and then `uint GetTargetVersion(Target, TargetVersion)` internally?  Or target-specific version structs `MslVersion(major, minor, patch)`, `GlslVersion(major, minor)`, or what have you?